src/common/d3Components/zoom/zoom.ts

```diff
@@ -0,0 +1,88 @@
+import { event as d3Event } from 'd3-selection';
+import { zoom, zoomIdentity } from 'd3-zoom';
+import { D3Selection, Extension } from '../../../common/types';
+
+interface ZoomArea {
+  id: any;
+}
+
+interface Props {
+  svg: D3Selection;
+  node: D3Selection;
+  zoomAreas: ZoomArea[];
+  nodeExtension: Extension;
+  nodeSelectionElement: string;
+  getZoomAreaExension: (zoomArea: ZoomArea) => Extension;
+  maxZoomScale: number;
+  clickZoomFitScale: number;
+  width: number;
+  height: number;
+}
+
+interface State {
+  currentZoomAreaId: any;
+}
+
+export const zoomComponent = (props: Props) => {
```

Comment:
Same here I would add the suffix, it can lead to confusions, you see this component in a page and you are not sure whether it's react or d3 at first sight.

zoomD3Component